store/store_kv.go

```diff
@@ -0,0 +1,140 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/portworx/kvdb"
+)
+
+const (
+	cloudDriveKey           = "clouddrive/"
+	cloudDriveLockKey       = "_lock"
+	maxWatchErrors          = 5
+	defaultLockTryDuration  = 1 * time.Minute
+	defaultLockHoldDuration = 3 * time.Minute
+)
+
+var (
+	// ErrKvdbNotInitialized is returned when kvdb is not initialized
+	ErrKvdbNotInitialized = errors.New("KVDB is not initialized")
+)
+
+type kvStore struct {
+	k                kvdb.Kvdb
+	keyName          string
+	lockTryDuration  time.Duration
+	lockHoldDuration time.Duration
+}
+
+// NewKVStore returns a Store implementation which is a wrapper over
+// kvdb.
+func NewKVStore(kv kvdb.Kvdb, name string) (Store, error) {
+	if len(name) == 0 {
```

Comment:
Lets fail the NewKVStore operation if name is empty and force the callers to set a key, so that we won't end up causing key conflicts